packages/create-app/src/lib/tasks.ts

```diff
@@ -236,3 +244,79 @@ export async function moveAppTask(
       });
   });
 }
+
+/**
+ * Read git configs by creating a temp folder and initializing a repo
+ *
+ * @throws if `exec` fails
+ */
+export async function readGitConfig(): Promise<GitConfig | undefined> {
+  const tempDir = resolvePath(os.tmpdir(), 'git-temp-dir');
+
+  const runCmd = (cmd: string) =>
+    exec(cmd).catch(error => {
+      process.stdout.write(error.stderr);
+      process.stdout.write(error.stdout);
+      throw new Error(`Could not execute command ${chalk.cyan(cmd)}`);
+    });
+
+  const isGitAvailable = await commandExists('git').catch(() => false);
+
+  if (!isGitAvailable) return;
+
+  try {
+    await fs.mkdir(tempDir);
+
+    process.chdir(tempDir);
```

Comment:
Regarding the e2e test failures, as safer way to do this is to pass the `cwd` option to `exec` instead, rather than having this global state being mutated.